src/vivarium_testing_utils/automated_validation/data_loader.py

```diff
@@ -108,16 +108,59 @@ def get_data(self, data_key: str, source: DataSource) -> Any:
     def upload_custom_data(self, data_key: str, data: pd.DataFrame) -> None:
         self._add_to_cache(data_key, DataSource.CUSTOM, data)
 
+    def cache_gbd_data(
+        self, data_key: str, data: pd.DataFrame, overwrite: bool = False
+    ) -> None:
+        """Upload or update a custom DataFrame or Series to the GBD context given by a data key."""
+        if overwrite:
+            if data_key in self._raw_data_cache[DataSource.GBD]:
+                del self._raw_data_cache[DataSource.GBD][data_key]
+        if data_key in self._raw_data_cache[DataSource.GBD] and not overwrite:
+            existing = self._raw_data_cache[DataSource.GBD][data_key]
+            if isinstance(existing, str):
+                raise ValueError(
+                    f"Existing GBD data for {data_key} is a string and cannot be updated without the overwrite flag set to True."
+                )
+            else:
+                if set(existing.index.names) != set(data.index.names):
+                    raise ValueError(
+                        f"Cannot update GBD data for {data_key} with different index names."
+                    )
+                if not existing.index.equals(data.index):
+                    # Check if the new data has non-overlapping indices with existing data
+                    overlapping_indices = existing.index.intersection(data.index)
+                    if len(overlapping_indices) > 0:
+                        raise ValueError(
```

Comment:
Is there a case to be made where we'd want to check if the values in the two datasets for the overlapping indices are identical and, if so, then don't raise?